lib/Varien/File/Uploader.php

```diff
@@ -505,8 +505,8 @@ private function _setUploadFileId($fileId)
             preg_match("/^(.*?)\[(.*?)\]$/", $fileId, $file);
 
             if (count($file) > 0
-                && (count($file[0]) > 0)
-                && (count($file[1]) > 0)
+                && isset($file[0])
+                && isset($file[1])
```

Comment:
is `isset()` a good enoug replacement here? If they used count() on a string here, then it was probably meant for the lengts, to avoid empty strings.
Might a `is_string() && strlen() > 0` be a better replacement? What other types can we expect here besides strings?

Reply:
> Might a `is_string() && strlen() > 0`

Shorter `$file[0] !== ''`?

Reply:
Good call. It seems like `preg_match` can potentially return integers as a match as well according to documentation examples, although some of the docs are contradictory. In the examples there are unquoted integers as matches, but the return type description states the `text` that is matched.

Regarding the logic: The 0-index is simply the string to be matched (if a match is present), so checking existence on that should be sufficient. The 1-index will be a match (or not set at all), so I believe checking existence should be sufficient. Although I don't see a guarantee that the match (if present) will necessarily have a length > 0, this is my assumption. In testing I'm unable to match a 'space' character either.

Reply:
`count()` when used on a string pre PHP 8.0, had the same effect of converting the value to an array before calling it. So `count('')` would still result in `1` because it's similar to `count([''])`.

From the context of the usage, I think the last two conditions are useless and can be removed since for this specific regex if `count($file) > 0`, then `$file[0]` and `$file[1]` are guaranteed to be set (empty string if they are not matched), and by changing it to `!== ''` we are changing the behavior.

Reply:
> If they used count() on a string here, then it was probably meant for the lengts, to avoid empty strings.

This would not work. In PHP7 `count()` on a string always returned int  `1` (not checking for length)

From next lines ....

```
$fileAttributes = $_FILES[$file[0]];
```

Array key `$file[0]` must be string or int. Neither int nor empty string make sense here, so i think @Flyingmana is right with string comparison. (`!== ''`) is just a shorter version for that)